src/features/foods/components/AddEditForm/AddDrawerBody.tsx

```diff
@@ -0,0 +1,132 @@
+  const [selectedContainer, setSelectedContainer] = useState('');
+
+  /**
+   * Process when a group is selected
+   * @param event
+   */
+  const handleSelectGroup = (value: string) => {
+    setSelectedGroup(value);
+    setSelectedContainer('');
+  };
+
+  /**
+   * Process when a container is selected
+   * @param event
+   */
+  const handleSelectContainer = (containerId: string) => {
+    setSelectedContainer(containerId);
+    setSelectedGroup(containerIdGroupIdMap[containerId]);
+  };
+
+  return (
+    <div className="flex flex-col gap-4">
+      <LabeledInput label="Name" htmlFor="name" required>
+        <Input name="name" id="name" placeholder="Container name" />
+      </LabeledInput>
+      <LabeledInput label="Group" htmlFor="group" required>
+        <Select
+          name="group"
+          value={selectedGroup}
+          onValueChange={(value: string) => handleSelectGroup(value)}
+        >
+          <SelectTrigger id="group">
+            <SelectValue placeholder="Select a group" />
+          </SelectTrigger>
+          <SelectContent>
+            {Object.keys(groupIdContainerIdsMap).map((group) => (
+              <SelectItem key={group} value={group}>
+                {groupIdNameMap[group]}
+              </SelectItem>
+            ))}
+          </SelectContent>
+        </Select>
+      </LabeledInput>
+      <LabeledInput label="Container" htmlFor="container" required>
+        <Select
+          name="container"
+          value={selectedContainer}
+          onValueChange={(value: string) => handleSelectContainer(value)}
+        >
+          <SelectTrigger id="container">
+            <SelectValue placeholder="Select a container" />
+          </SelectTrigger>
+          <SelectContent>
+            {Object.entries(groupIdContainerIdsMap).map(
+              ([groupId, containers]: [string, string[]], i) => {
+                return (
+                  <SelectGroup key={i}>
+                    <SelectLabel>{groupIdNameMap[groupId]}</SelectLabel>
+                    {containers.map((container) => (
+                      <SelectItem key={container} value={container}>
+                        {containerIdNameMap[container]}
+                      </SelectItem>
+                    ))}
+                  </SelectGroup>
+                );
+              },
+            )}
+          </SelectContent>
+        </Select>
+      </LabeledInput>
+      <div className="grid grid-cols-2 gap-6">
+        <LabeledInput label="Quantity" htmlFor="quantity">
+          <Input name="quantity" id="quantity" type="number" />
+        </LabeledInput>
+        <LabeledInput label="Unit" htmlFor="unit">
+          <Input name="unit" id="unit" type="text" />
+        </LabeledInput>
+      </div>
+      <LabeledInput label="Expiry" htmlFor="expiry">
+        <FormDatePicker
+          id="expiry"
+          hiddenInput={(date) => <Input type="hidden" name="expiry" value={date} />}
```

Comment:
This component applies the technique called [render prop(s)](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop).

I implemented this in order to define a state and component together in the child component while making the state accessible from the parent component.

However, this approach seems to be getting less common. Thus, I may replace it with the HOC or custom hook shortly.